src/Sarif/Core/FileData.cs

```diff
@@ -15,6 +15,8 @@ namespace Microsoft.CodeAnalysis.Sarif
     /// </summary>
     public partial class FileData : ISarifNode
     {
+        internal static IFileSystem _fileSystem = new FileSystem();
```

Comment:
>internal static IFileSystem _fileSystem = new FileSystem() [](start = 8, length = 58)

I suggest that you use constructor injection, as follows (well, it's not exactly "constructor" injection, but close enough):
```
public static FileData Create(TheArgs theArgs)
{
    Create(theArgs, new FileSystem());
}

// For testing only:
internal static FileData Create(TheArgs theArgs, IFileSystem fileSystem)
{
    // the real code.
}
```

Reply:
why should I instantiate the file system for each call? Also, you approach will require mucking with every API that utilizes the file system object. my approach modifies a common internal member that arbitrary API could use.

---
In reply to: [136627741](https://github.com/Microsoft/sarif-sdk/pull/614#discussion_r136627741) [](ancestors = 136627741)

Reply:
Because global data is evil. You could imagine getting a test wrong by neglecting-to-set-up-a-new-mock-and-assign-it-to-the-static; then your test would accidentally use the mock from some previous test -- and who could tell which one? -- and your test would fail unpredictably.

With injection, you guarantee each test to use its own mock.

---
In reply to: [136633228](https://github.com/Microsoft/sarif-sdk/pull/614#discussion_r136633228) [](ancestors = 136633228,136627741)

Reply:
As we discussed, better pattern is:
```
public static FileData Create(TheArgs theArgs, IFileSystem fileSystem = null)
{
    if (fileSystem == null) fileSystem = new FileSystem();
}
```

---
In reply to: [136634455](https://github.com/Microsoft/sarif-sdk/pull/614#discussion_r136634455) [](ancestors = 136634455,136633228,136627741)

Reply:
beautiful, like it

---
In reply to: [136639877](https://github.com/Microsoft/sarif-sdk/pull/614#discussion_r136639877) [](ancestors = 136639877,136634455,136633228,136627741)